### ISSUE TITLE
add status changing button for aborted and force-complete

### ISSUE THREAD

**Issue by ticoann** [dmwm/WMCore#5702]
Ops also able to change request status to aborted and force-complete from a appropriate status.
I don't think there need to be a bulk update but I might be wrong.

**Comment by vkuznet** [comment]
Seangchan, could you please elaborate more on this. The page should show aborted/force-complete workflows and then for every workflow they need a changing button? What this action should do, i.e. change from aborted/force-complete to what?

**Comment by ticoann** [comment]
Sorry Valentin being not clear. I meant from any allowed previous state, the workflow should be able to move to aborted or force-completed. 
Basically "assigned-approved", "assigned", "aborted", "force-completed", "closed-out", "announced" those are currently manual transition status  (ops do it, not the system). So they should be able to have some interface to change the status. 
"assigned",  "closed-out", "announced"  - These status changes need to b automated. But I think for aborted and force-completed need the ops intervention.
As far as GUI, currently we do one by one. Meaning ops need to know which workflow they need to abort or force completed.  (Basically they look at the wmstats and decide which workflow to abort and wmstats provide the link to workflow in reqmgr page).

I am not sure whether bulk display is needed. (We can discuss in more detail later).

**Comment by vkuznet** [comment]
Ok, let me rephrase. Data-ops need an ability to apply either "aborted" or "force-completed" action to any workflow. This should be done by workflow basis. They will know on which workflow they'll need to perform either action. In other words, wmstats will provide a link: https://cmsweb-testbed.cern.ch/reqmgr2/fetch?rid=request-sryu_ReqMgr2_test_150318_181019_2305 then this page should contain a "abort" or "force-compete" buttons. Please confirm.

**Comment by amaltaro** [comment]
That's right Valentin. CompOps knows exactly what needs to be aborted or force-completed, thus the workflow needs to provide a select box with the available options (allowed transitions), e.g.:
https://cmsweb-testbed.cern.ch/reqmgr/view/details/amaltaro_ReRecoSkim_HG1504a_Ag100_150324_111331_8040

where you can see "acquired" and then a select box with those allowed transitions that comes from 
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py

**Comment by vkuznet** [comment]
Alan, I want to double check one more thing. The requests statuses are listed in two different files, the one you provided (https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/RequestManager/RequestDB/Settings/RequestStatus.py) and here (https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/DataStructs/RequestStatus.py). Which status definition should I use? I can either use NextStatus from file you pointed out or I can use REQUEST_STATE_TRANSITION from later file. My understanding that it should be later one, but I want double check with you/Seangchan.

**Comment by ticoann** [comment]
Valentin, yes always later one. First one will be removed when reqmgr is retired.

**Comment by amaltaro** [comment]
Thanks Seangchan! I thought it was the other way around. Why I have to always confuse these mappings... :-(

**Comment by vkuznet** [comment]
Ok, here is initial prototype, see attachment. For the form itself, I just set RequestName/RequestStatus and pass it to JS function like that:
```
function ChangeRequestStatus() {
    var parameters = $('change-status-form').serialize(true);
    ajaxRequest('$base/data/request', parameters);
}
```

Seangchan, would it be sufficient for back-end API? My understanding that I just need to pass parameters (RequestName and RequestStatus) and back-end API will modify workflow status.
![screen shot 2015-03-24 at mar 24 12 11 26 pm](https://cloud.githubusercontent.com/assets/125138/6806475/565db474-d21f-11e4-911f-06fcdf113a92.png)

**Comment by ticoann** [comment]
> Seangchan, would it be sufficient for back-end API? My understanding that I just need to pass parameters (RequestName and RequestStatus) and back-end API will modify workflow status.

Yes, that would be sufficient. FYI, ReqMgr2 is strict about what can be updated in certain state. Only exception is status change.

**Comment by vkuznet** [comment]
Ok, please take PR: https://github.com/dmwm/WMCore/pull/5709 with these changes. I only tested web UI part, but didn't test back-end API callback (I don't have proper frontend setup on my laptop, so the test should be done on reqmgr-dev2 node).

**Comment by vkuznet** [comment]
Once I tested PR #5709, I found that prototype library can't handle properly PUT requests required to update the status. I reorganized code and used internal method to make PUT request, see PR https://github.com/dmwm/WMCore/pull/5720

**Comment by vkuznet** [comment]
New changes (change from prototype to jQuery) are here: https://github.com/dmwm/WMCore/pull/5724
